notebooks/DATB/spectral_registration_manual/Spectral Cube Reprojection and Alignment by Manual Feature Selection.ipynb

```diff
@@ -0,0 +1,404 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ifu1_data, ifu1_wcs = ifu1.data(sl)\n",
+    "ifu2_data, ifu2_wcs = ifu2.data(sl)\n",
+    "ifu3_data, ifu3_wcs = ifu3.data(sl)\n",
+    "ifu4_data, ifu4_wcs = ifu4.data(sl)\n",
+    "\n",
+    "ifu1_data_reproj = ifu1_data\n",
+    "ifu2_data_reproj, footprint2 = reproject_interp((ifu2_data, ifu2_wcs), ifu1_wcs, ifu1_data.shape)\n",
+    "ifu3_data_reproj, footprint3 = reproject_interp((ifu3_data, ifu3_wcs), ifu1_wcs, ifu1_data.shape)\n",
+    "ifu4_data_reproj, footprint4 = reproject_interp((ifu4_data, ifu4_wcs), ifu1_wcs, ifu1_data.shape)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(2,  figsize=[10,8])\n",
+    "plt.clf()\n",
+    "plt.subplot(221, projection=ifu1_wcs)\n",
+    "show(ifu1_data, 'Sequence 1')\n",
+    "\n",
+    "plt.subplot(222, projection=ifu1_wcs)\n",
+    "show(ifu2_data_reproj, 'Sequence 2 - Reprojected')\n",
+    "\n",
+    "plt.subplot(223, projection=ifu1_wcs)\n",
+    "show(ifu3_data_reproj, 'Sequence 3 - Reprojected')\n",
+    "\n",
+    "plt.subplot(224, projection=ifu1_wcs)\n",
+    "show(ifu4_data_reproj, 'Sequence 4 - Reprojected')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So, as expected, Sequence 1 and 2 are still centered at the correct location, but Sequence 3 and 4 are still shifted slightly lower and to the right.  But, at this point, all the images are in the same WCS frame (Sequence 1)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Alignment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This alignment uses mouse clicks on the image to determine how much to shift images 2 and on in order to map it back to the first image.\n",
+    "\n",
+    "There are two methods used in the mapping, one is an image \"roll\" based on numpy.roll and the second is is an interpoliation based on scipy.ndimage.interplation.shift."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**How to use this**\n",
+    "\n",
+    "Click N points in the first image on specific features. Then when done click on the right mouse button.\n",
+    "\n",
+    "In the subsequent images, click the **same** features and when done click on the right mouse button.\n",
+    "\n",
+    "Once the clicking is done on the last image go to the next notebook cell and run it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "images = [ifu1_data_reproj, ifu2_data_reproj, ifu3_data_reproj, ifu4_data_reproj]\n",
+    "dg = DisplayGinput(images)"
```

Comment:
Instead of having `DisplayGinput.__init__` start the interaction, perhaps there should be a separate method called something like `DisplayGinput.show_and_register_images()`? In its default mode, that would do `show_get` and hook up the callback.   But it would have an alternate mode that takes one argument which is the set of points for registration.  Then it would *not* hook of the callback, but just show the static plot. 

If you do that, then you can leave this cell alone, but add a cell with:

```
#dg.show_and_register_images() # uncomment this line and comment the next to get interactivity
dg.show_and_register_images([a,set,of,points])
```
which will have it full execute in "non-interactive" mode, but the user can do the interactive mode when they've downloaded the notebook